### PR TITLE
Release 0.7.1 

### PR DESCRIPTION
**Pull request by sakows**
Update nginmesh to be compatible with Istio 0.7.1
1. Remove Kafka streaming module
2. Update agent with Mixer module 0.7.1 version
3. Add Ingenious Sample Application
4. Update E2E Test Cases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,69 +110,40 @@ reviews-v1-874083890-f0qf0                  2/2       Running   0          6m
...
-
 ```
 kubectl get ingress -o wide       
 ```
-
 6. Open the Bookinfo application in a browser using the following link:
 ```
 http://<Public-IP-of-the-Ingress-Controller>/productpage
 ```
+Note 1: For E2E routing rules and performace testing you could refer to [E2E Test](istio/tests/README.md).
```
Comment (pleshakov) [inline README.md]: please remove this two notes

File: README.md [hunk 2]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
 ### Prerequisites
 Make sure you have a Kubernetes cluster with at least 1.9 or greater due to fact only automatic sidecar injection is supported and no alpha feature is enabled. Please see [Prerequisites](https://istio.io/docs/setup/kubernetes/quick-start.html) for setting up a kubernetes cluster.
```
Comment (pleshakov) [inline README.md]: make sure we reference the docs from the 0.7 release of Istio

Comment (pleshakov) [inline README.md]: `Make sure you have a Kubernetes cluster with at least 1.9 or greater due to fact only automatic sidecar injection is supported and no alpha feature is enabled.` -> `Make sure you have a cluster with Kubernetes 1.9 or newer.`

File: README.md [hunk 3]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
...
 
-### Installing Istio and nginMesh
+### Install Istio and nginMesh
 nginMesh requires installation of Istio first.
 
 1. Download and install Istio 0.7.1:
 ```
```
Comment (pleshakov) [inline README.md]: Please indent all paragraphs with commands. For example:

Before:
1. Download and install Istio 0.7.1:
 ```
curl -L https://git.io/getLatestIstio | ISTIO_VERSION=0.7.1 sh -
```
After:
1. Download and install Istio 0.7.1:
    ```
   curl -L https://git.io/getLatestIstio | ISTIO_VERSION=0.7.1 sh -
   ```

File: README.md [hunk 4]
```diff
@@ -9,7 +9,6 @@ The current version of nginMesh is designed to work with Istio release 0.7.1. It
 
 ## Architecture
 The diagram below depicts how an NGINX sidecar proxy is implemented. Sidecar uses the open source version of NGINX compiled with modules for tracing and monitoring.
```
Comment (pleshakov) [inline README.md]: ` Sidecar uses` -> `The  sidecar uses`

File: README.md [hunk 5]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
...
-  istio-ingress   10.83.245.171   35.184.245.62     80:32730/TCP,443:30574/TCP    5h
-  istio-pilot     10.83.251.173   <none>            8080/TCP,8081/TCP             5h
-  istio-mixer     10.83.244.253   <none>            9091/TCP,9094/TCP,42422/TCP   5h
+istio-ingress            LoadBalancer   10.47.252.40    35.237.173.47   80:32171/TCP,443:32198/TCP                   19h
+istio-mixer              ClusterIP      10.47.251.225   <none>          9091/TCP,15004/TCP,9093/TCP,9094/TCP,9102/TCP,9125/UDP,42422/TCP    19h
+istio-pilot              ClusterIP      10.47.254.118   <none>          15003/TCP,15005/TCP,15007/TCP,15010/TCP,8080/TCP,9093/TCP,443/TCP   19h
+istio-sidecar-injector   ClusterIP      10.47.242.139   <none>          443/TCP                                       9h
 ```
-
 5. Ensure the following Kubernetes pods are up and running: istio-pilot-* , istio-mixer-* , istio-ingress-*  and istio-initializer-* :
```
Comment (pleshakov) [inline README.md]: in all steps make sure there is no space between the last of word of a sentence and the colon. For example:
Before:
`and istio-initializer-* :`
After:
`and istio-initializer-*:`

File: README.md [hunk 6]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
...
 curl -L https://git.io/getLatestIstio | ISTIO_VERSION=0.7.1 sh -
 ```
-2. Download nginMesh release 0.7.2:
+2. Download nginMesh release 0.7.1:
 ```
-curl -L https://github.com/nginmesh/nginmesh/releases/download/0.7.2/nginmesh-0.7.2.tar.gz | tar zx
+curl -L https://github.com/nginmesh/nginmesh/releases/download/0.7.1/nginmesh-0.7.1.tar.gz | tar zx
 ```
-
 3. Deploy Istio between sidecars:
```
Comment (pleshakov) [inline README.md]: `Deploy Istio between sidecars` ->  `Deploy Istio:`

File: README.md [hunk 7]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
...
-  istio-pilot-2275554717-93c43        1/1       Running   0          5h
-  istio-mixer-2104784889-20rm8        2/2       Running   0          5h
+istio-ca-86f55cc46f-nprhw                1/1       Running   0          19h
+istio-ingress-5bb556fcbf-c7tgt           1/1       Running   0          19h
+istio-mixer-86f5df6997-fvzjx             3/3       Running   0          19h
+istio-pilot-67d6ddbdf6-xhztz             2/2       Running   0          19h
+istio-sidecar-injector-5b8c78fd6-8dvq6   1/1       Running   0          9h
 ```
-
 6. Automatic sidecar:
```
Comment (pleshakov) [inline README.md]: `Automatic sidecar:` -> `Enable automatic sidecar injection`
Remove `To set up sidecar injection, please run following script which will install Istio webhook with nginMesh customization.`

File: README.md [hunk 8]
```diff
@@ -21,53 +20,47 @@ Below are instructions to quickly install and configure nginMesh.  Currently, on
...
 To set up sidecar injection, please run following script which will install Istio webhook with nginMesh customization.
 ```
-nginmesh-0.7.2/install/kubernetes/install-sidecar.sh
+nginmesh-0.7.1/install/kubernetes/install-sidecar.sh
 ```
-
 7. Verify that istio-injection label is not labeled for the default namespace :
```
Comment (pleshakov) [inline README.md]: `Verify that istio-injection label is not labeled for the default namespace :` -> `Verify that the istio-injection label is not applied to the default namespace :`

File: README.md [hunk 9]
```diff
@@ -192,69 +110,40 @@ reviews-v1-874083890-f0qf0                  2/2       Running   0          6m
...
-2. Delete Tiller deployment:
-
-```
-kubectl delete deployment tiller-deploy --namespace kube-system
-```
-
- 
-
-
 ## Limitations
```
Comment (pleshakov) [inline README.md]: Add `The latest supported version of Istio is 0.7` to the list of limitations.

File: istio/agent/pilot/client_test.go [hunk 10]
```diff
@@ -11,7 +11,7 @@ import (
 func TestUnmarshal(t *testing.T) {
```
Comment (pleshakov) [inline istio/agent/pilot/client_test.go:11]: Please fix this test

File: istio/release/Makefile [hunk 11]
```diff
@@ -1,4 +1,4 @@
-VERSION=0.7.2
+VERSION=0.8.0
```
Comment (pleshakov) [inline istio/release/Makefile]: version if not correct

File: istio/release/install/kubernetes/templates/nginmesh-sidecar-injector-configmap.yaml.tmpl [hunk 12]
```diff
@@ -39,7 +39,7 @@ data:
           privileged: true
       containers:
       - name: istio-proxy
-        image: {NGINMESH_IMAGE_HUB}/nginmesh/proxy_debug:{NGINMESH_VERSION}
+        image: nginxinc/nginmesh_proxy_debug:0.7.1
```
Comment (pleshakov) [inline istio/release/install/kubernetes/templates/nginmesh-sidecar-injector-configmap.yaml.tmpl]: the image is hardcoded. please fix

File: istio/release/samples/ingenious/kube/README.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
+# Ingenious
```
Comment (pleshakov) [inline istio/release/samples/ingenious/kube/README.md]: Please remove the Ingenious app example

File: istio/tests/Makefile [hunk 14]
```diff
@@ -1,5 +1,5 @@
 KUBE_VERSION=v1.9.4
-NGINMESH_VERSION=0.6.0
+NGINMESH_VERSION=0.8.0
```
Comment (pleshakov) [inline istio/tests/Makefile]: the version if not correct

**Review by pleshakov — CHANGES_REQUESTED** [review]
Please see my suggestions

File: README.md (hunk 1 above)
Comment (sakows) [reply README.md]: ok

File: README.md (hunk 2 above)
Comment (sakows) [reply README.md]: ok

Comment (sakows) [reply README.md]: ok

File: README.md (hunk 3 above)
Comment (sakows) [reply README.md]: ok

File: istio/release/samples/ingenious/kube/README.md (hunk 13 above)
Comment (sakows) [reply istio/release/samples/ingenious/kube/README.md]: ok

File: istio/release/Makefile (hunk 11 above)
Comment (sakows) [reply istio/release/Makefile]: ok

File: README.md (hunk 4 above)
Comment (sakows) [reply README.md]: ok

File: istio/tests/Makefile (hunk 14 above)
Comment (sakows) [reply istio/tests/Makefile]: ok

File: istio/agent/pilot/client_test.go (hunk 10 above)
Comment (sakows) [reply istio/agent/pilot/client_test.go:11]: ok

File: README.md (hunk 6 above)
Comment (sakows) [reply README.md]: ok

File: README.md (hunk 5 above)
Comment (sakows) [reply README.md]: ok

File: README.md (hunk 7 above)
Comment (sakows) [reply README.md]: ok

File: README.md (hunk 8 above)
Comment (sakows) [reply README.md]: ok

**Review by pleshakov — APPROVED** [review]

**Pull request merged**